doc/APIScope.md

```diff
@@ -0,0 +1,94 @@
+# API Scope
+
+This document describes the scope of the Vitess APIs, and how they map to
+traditional concepts like database and tables.
+
+## Introduction
+
+Vitess is exposed as a single database by clients, but can be composed of any
+arbitrary number of databases, some of them possibly sharded with large numbers
+of shards. It is not obvious to map this system with clients that expect only a
+single database.
+
+## Shard Names and Key Ranges
+
+For unsharded keyspaces, or custom sharded keyspaces, the shard names used are
+usually just a number, like `0`.
```

Comment:
The `0` shard is a special case which will always create confusion. Hence I suggest to better call it out e.g.

> For unsharded keyspaces, or custom sharded keyspaces, the  shard names have traditionally been numbers e.g.  `0`. (It is important to note that Vitess **does not** interpret these shards (including shard name `0`) as range-based shard.)